samples/bluetooth/central_iso/src/main.c

```diff
@@ -0,0 +1,244 @@
+
+static void start_scan(void);
+
+static struct bt_conn *default_conn;
+static struct k_delayed_work iso_send_work;
+static struct bt_iso_chan iso_chan;
+NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU, NULL);
+
+/**
+ * @brief Send ISO data on timeout
+ *
+ * This will send an increasing amount of ISO data, starting from 1 octet.
+ *
+ * First iteration : 0x00
+ * Second iteration: 0x00 0x01
+ * Third iteration : 0x00 0x01 0x02
+ *
+ * And so on, until it wraps around the configured ISO TX MTU (CONFIG_BT_ISO_TX_MTU)
+ *
+ * @param work Pointer to the work structure
+ */
+static void iso_timer_timeout(struct k_work *work)
+{
+	int ret;
+	static uint8_t buf_data[MAX_ISO_APP_DATA];
+	static bool data_initialized;
+	struct net_buf *buf;
+	static size_t len_to_send = 1;
+
+	if (!data_initialized) {
+		for (int i = 0; i < ARRAY_SIZE(buf_data); i++) {
+			buf_data[i] = (uint8_t)i;
+		}
+	}
+
+	buf = net_buf_alloc(&tx_pool, K_FOREVER);
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+
+	net_buf_add_mem(buf, buf_data, len_to_send);
+
+	ret = bt_iso_chan_send(&iso_chan, buf);
+
+	if (ret < 0) {
+		printk("Failed to send ISO data (%d)\n", ret);
+	}
+
+	k_delayed_work_submit(&iso_send_work, K_MSEC(1000));
+
+	len_to_send++;
+	if (len_to_send > ARRAY_SIZE(buf_data)) {
+		len_to_send = 1;
+	}
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	if (default_conn) {
+		return;
+	}
+
+	/* We're only interested in connectable events */
+	if (type != BT_GAP_ADV_TYPE_ADV_IND && type != BT_GAP_ADV_TYPE_ADV_DIRECT_IND) {
+		return;
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	printk("Device found: %s (RSSI %d)\n", addr_str, rssi);
+
+	/* connect only to devices in close proximity */
+	if (rssi < -70) {
+		return;
+	}
+
+	if (bt_le_scan_stop()) {
+		return;
+	}
+
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_DEFAULT,
+				&default_conn);
+	if (err) {
+		printk("Create conn to %s failed (%u)\n", addr_str, err);
+		start_scan();
+	}
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* This demo doesn't require active scan */
+	err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+	if (err) {
+		printk("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void iso_connected(struct bt_iso_chan *chan)
+{
+	printk("ISO Channel %p connected\n", chan);
+
+	/* Start send timer */
+	k_delayed_work_submit(&iso_send_work, K_MSEC(0));
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan)
+{
+	printk("ISO Channel %p disconnected\n", chan);
+	k_delayed_work_cancel(&iso_send_work);
+}
+
+static struct bt_iso_chan_ops iso_ops = {
+	.connected	= iso_connected,
+	.disconnected	= iso_disconnected,
+};
+
+static struct bt_iso_chan_qos iso_qos = {
+	.dir = BT_ISO_CHAN_QOS_OUT,
+	.interval = 1000,
```

Comment:
1 ms?

Reply:
Changed to 10ms. Luiz suggested to keep it as the same value as the timer, but I figured that it might be nice to see some of the empty packets as well.